R/utilities-simulation-results.R

```diff
@@ -34,15 +34,23 @@ getOutputValues <- function(simulationResults,
   quantitiesOrPaths <- quantitiesOrPaths %||% simulationResults$allQuantityPaths
   quantitiesOrPaths <- c(quantitiesOrPaths)
 
-  # If quantities are passed, get their paths.
-  paths <- quantitiesOrPaths
-  if (isOfType(paths, Quantity)) {
-    paths <- unlist(lapply(paths, function(x) x$path))
+  if (length(quantitiesOrPaths) == 0) {
+    return(list(data = NULL, metaData = NULL))
   }
-  paths <- unique(paths)
 
-  if (length(paths) == 0) {
-    return(list(data = NULL, metaData = NULL))
+  # If quantities are passed, get their paths.
+  if (isOfType(quantitiesOrPaths, Quantity)) {
+    quantities <- uniqueEntities(quantitiesOrPaths)
+    paths <- unlist(
+      lapply(quantities, function(x) x$path)
+      )
+    names(quantities) <- paths

```

Comment:
That can be moved outside of if then since you are doing it everywhere

Reply:
Done.